From 02c0226133d9fe103bbb5bdec70fc17094c55fb3 Mon Sep 17 00:00:00 2001
From: Austin Gill <Notgnoshi@gmail.com>
Date: Fri, 7 May 2021 17:20:18 -0500
Subject: [PATCH] Add missing environment variable for config file

Most of the gitlint command-line options have corresponding environment
variables following the format GITLINT_<commandline_option>, added in
69aa5ffaeb7dae849f8ab655ea0426e26deee44f.
---
 docs/configuration.md | 14 ++++++++++++++
 gitlint/cli.py        |  3 ++-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/docs/configuration.md b/docs/configuration.md
index 50c4e63..b1e64b0 100644
--- a/docs/configuration.md
+++ b/docs/configuration.md
@@ -297,6 +297,20 @@ GITLINT_TARGET=/home/joe/myrepo/ gitlint     # using env variable
 [general]
 target=/home/joe/myrepo/
 ```
+### config
+
+Path where gitlint looks for a config file.
+
+Default value              |  gitlint version | commandline flag  | environment variable   
+---------------------------|------------------|-------------------|-----------------------
+ (empty)                   | >= 0.16.0        | `--config`        | `GITLINT_CONFIG`
+
+#### Examples
+```sh
+gitlint --config=~/.config/gitlint/gitlint.ini
+gitlint -C ~/.gitlint
+GITLINT_CONFIG=~/.gitlint gitlint
+```
 
 ### extra-path
 
diff --git a/gitlint/cli.py b/gitlint/cli.py
index 9b16d47..663445f 100644
--- a/gitlint/cli.py
+++ b/gitlint/cli.py
@@ -200,7 +200,8 @@ def __init__(self, config, config_builder, refspec, msg_filename, gitcontext=Non
 @click.option('--target', envvar='GITLINT_TARGET',
               type=click.Path(exists=True, resolve_path=True, file_okay=False, readable=True),
               help="Path of the target git repository. [default: current working directory]")
-@click.option('-C', '--config', type=click.Path(exists=True, dir_okay=False, readable=True, resolve_path=True),
+@click.option('-C', '--config', envvar='GITLINT_CONFIG',
+              type=click.Path(exists=True, dir_okay=False, readable=True, resolve_path=True),
               help=f"Config file location [default: {DEFAULT_CONFIG_FILE}]")
 @click.option('-c', multiple=True,
               help="Config flags in format <rule>.<option>=<value> (e.g.: -c T1.line-length=80). " +
